services/meta/write_authorizer.go

```diff
@@ -0,0 +1,27 @@
+package meta
+
+import (
+	"fmt"
+
+	"github.com/influxdata/influxdb/influxql"
+)
+
+type WriteAuthorizer struct {
+	Client *Client
+}
+
+func NewWriteAuthorizer(c *Client) *WriteAuthorizer {
+	return &WriteAuthorizer{Client: c}
+}
+
+// AuthorizeWrite returns nil if the user has permission to write to the database.
+func (a WriteAuthorizer) AuthorizeWrite(username, database string) error {
+	u, err := a.Client.User(username)
```

Comment:
I would possibly separate out a client error from an authorisation error, depending on the error possibilities from `Client.User`